relay/relay.go

```diff
@@ -344,6 +344,9 @@ func (r *Relay) tryRecoverLatestFile(parser2 *parser.Parser) error {
 		if result.Recovered {
 			r.tctx.L().Warn("relay log file recovered",
 				zap.Stringer("from position", latestPos), zap.Stringer("to position", result.LatestPos), log.WrapStringerField("from GTID set", latestGTID), log.WrapStringerField("to GTID set", result.LatestGTIDs))
+			if err = latestGTID.Truncate(result.LatestGTIDs); err != nil {
+				return err
+			}
 			err = r.meta.Save(result.LatestPos, result.LatestGTIDs)
```

Comment:
should save meta with `LatestGTID`?

Reply:
you are right! thanks a lot. done in https://github.com/pingcap/dm/pull/335/commits/9611a0c1c71645fe6b77aaaadb1ea19c1136e7b7.